roachtest: kill nodes at end of rebalancing test

tbg opened the pull request:
This is a high-level regression test for #32949. Prior to that PR, the
test would end up with an unavailable cluster.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/33059)
<!-- Reviewable:end -->

tbg committed:
roachtest: kill nodes at end of rebalancing test

This is a high-level regression test for #32949. Prior to that PR, the
test would end up with an unavailable cluster.

Release note: None

tbg commented:
@petermattis this is obsolete with your incoming PR https://github.com/cockroachdb/cockroach/pull/34126 and the roachtest within, correct?

petermattis commented:
> @petermattis this is obsolete with your incoming PR #34126 and the roachtest within, correct?

Yep. Somehow this slipped past my radar. I wasn't aware I was duplicating work with the testing in #34126.

tbg commented:
I'm much happier with your test, so I'm glad you weren't aware.
